moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -612,34 +615,55 @@ class KinematicsBase
   template <typename T>
   inline bool lookupParam(const std::string& param, T& val, const T& default_val) const
   {
-    ros::NodeHandle pnh("~");
-    if (pnh.hasParam(group_name_ + "/" + param))
-    {
-      val = pnh.param(group_name_ + "/" + param, default_val);
-      return true;
-    }
+    auto node = rclcpp::Node::make_shared("lookup_param");
 
-    if (pnh.hasParam(param))
+    auto parameters_lookup = std::make_shared<rclcpp::SyncParametersClient>(node);
```

Comment:
```suggestion
    rclcpp::SyncParametersClient parameters_lookup(node);
```
I don't see a reason for using a shared pointer here.

Reply:
Thanks for the input @henningkayser. While I don't fully agree in this case (I think it's pretty obvious from the constructor) I've accepted your suggestion in a commit that required further changes 7f706b5cbc17b8f48f223beaf3ade7615733396f